src/components/scene/pool.js

```diff
@@ -0,0 +1,65 @@
+var debug = require('../../utils/debug');
+var register = require('../../core/component').registerComponent;
+
+var warn = debug('components:pool:warn');
+
+module.exports.Component = register('pool', {
+  schema: {
+    mixin: {default: ''},
+    size: {default: 0}
+  },
+
+  multiple: true,
+
+  init: function () {
+    this.createEntities();
+  },
+
+  createEntities: function () {
+    var el;
+    var mixin = this.data.mixin;
+    if (!mixin) { return; }
+    this.pool = [];
+    this.pooledEls = [];
+    for (var i = 0; i < this.data.size; ++i) {
+      el = document.createElement('a-entity');
```

Comment:
Takes this part out from the loop to a another function `createEntity/pushNewEntity/...`, so it could be reused on `requestEntity` when the previous described param is set to true and there's no available entity